### DISCUSSION TITLE
Releasing 3.2

### DISCUSSION THREAD

**Discussion by dvarrazzo** [psycopg/psycopg#838] [General]
Hello,

psycopg 3.2 is pretty much ready now.

I will not have a computer available for the next week, therefore I think I would be better to avoid releasing it just now. I can make a release next Sunday 23rd of June or thereabout.

@dlax is there anything else we might want to address or do you think everything is ready?

Thank you very much to everything who have contributed to this work!

**Comment by dvarrazzo** [comment]
Of course, the CI has just decided to bitrot and building the next binary package is going to be interesting... https://github.com/psycopg/psycopg/actions/runs/9534293077/job/26278541052

**Comment by dlax** [comment]
> @dlax is there anything else we might want to address or do you think everything is ready?

No; it's ready as far as I am concerned.

Let me know if I can help with anything.

**Reply by dvarrazzo** [reply]
I didn't notice that [the binary package build CI was broken for a few weeks](https://github.com/psycopg/psycopg/actions/workflows/packages-bin.yml?query=branch%3Amaster). It seems that the first breakage in a scheduled run is after the CI refactoring in #824, but the OP had working CI runs on their repository (now gone).

I am now running [a build](https://github.com/psycopg/psycopg/actions/runs/9556071567) on a commit before the #824 merge to figure out if there is something wrong in that changeset or if it's the environment that changed and broke the CI.

**Reply by dvarrazzo** [reply]
Hmm... yesterday's run shows that CI wasn't broken before. So there is some difference in a new version of cibuildwheel or of the runners.

@stumpylog are you able to figure out what is currently broken in CI master? CI works as expected just before merging your changeset, in 179743e36126e1c380a5a22325a97e9ee4880e13.

**Reply by stumpylog** [reply]
Very odd, everything was green. I can try to look later today, but if it's urgent, I doubt there's anything in the newest cibuildwheel that is really needed

**Comment by dvarrazzo** [comment]
On Tue, 18 Jun 2024, 15:17 Trenton H, ***@***.***> wrote:

> Very odd, everything was green.
>
Indeed, I saw your builds. Very strange.

I can try to look later today, but if it's urgent, I doubt there's anything
> in the newest cibuildwheel that is really needed
>
Not urgent, but if it you are able to fix it by next week it would be a
very welcome help :)
Cheers

-- Daniele

**Reply by stumpylog** [reply]
Not a solution yet, but I thought those errors were familiar.  Looks like #830 took `cibuildwheel` to a newer version.  

In particular, Musllinux now defaults to musllinux_1_2 instead of musllinux_1_1.  That would be the source of problems there.  It appears a depends was removed from libpq.  That is hopefully an easy fix.

I suspect the MacOS errors are also from the upgrade, which mentions "Document the new delocate error checking macOS versions", and the error is a `DelocationError`.  The [FAQ](https://cibuildwheel.pypa.io/en/stable/faq/#macos-library-dependencies-do-not-satisfy-target-macos) though goes beyond my knowledge

**Reply by stumpylog** [reply]
I've pretty much struck out on the Mac builds.  I don't have any knowledge of mac development, so my shots in the dark aren't landing.  If anyone has further suggestions about the `DelocationError` and minimum targets, I'm happy to try them out.

If this remains an issue by next week, I would move back to 2.17.0 for the moment.

**Comment by dvarrazzo** [comment]
On Wed, 19 Jun 2024, 04:47 Trenton H, ***@***.***> wrote:

> I've pretty much struck out on the Mac builds. [...]
>
> If this remains an issue by next week, I would move back to 2.17.0 for the
> moment.
>
Thank you very much, your help is very appreciated. Let's see how things
evolve (or regress) early next week.

Cheers

-- Daniele

**Comment by dvarrazzo** [comment]
I should have fixed the build problems, let's see if [this job](https://github.com/psycopg/psycopg/actions/runs/9636804588) completes (it also carries libs upgrades, and openssl 1.1.1w -> 3.3.1 might be a risk).

The macOS build was fixed in ae6607a5344a9489c121afb3aca686cc11e0bd0b by adding the `MACOSX_DEPLOYMENT_TARGET` vars. The musllinux build was fixed in ae9bf53b043d73765ee4f3215eb0b3ee4f5758c5 by adding the missing dependency (the original error was more confusing as the github artifact cache was getting in the way).

@dlax, considering that postgres v17 is in beta, does it make sense to release psycopg 3.2 with binary packages bundling libpq 16? I'm ok either waiting for v17 or releasing as soon as all the chores are done and then releasing 3.2.1 (or greater) as soon as v17 is released, to enable all the features it brings.

**Reply by dlax** [reply]
My preference goes for releasing 3.2 as soon as possible and then expose v17 features (along with docs update, like https://github.com/psycopg/psycopg/pull/774) in a later 3.2.x release when Postgres itself gets released (~September?).

**Comment by dvarrazzo** [comment]
Psycopg 3.2 packaged released! Writing down a few announcements to send around tomorrow.

Thank you very much! :clinking_glasses:
